Update ACS to 4.6 and enable scannerV4

Roming22 opened the pull request:
Recommended configuration after reaching out the ACS team regarding [RHTAPBUGS-1316](https://issues.redhat.com//browse/RHTAPBUGS-1316)

Roming22 commented:
/retest

xinredhat commented:
/retest

Roming22 committed:
feat: Upgrade to ACS 4.6

rh-pre-commit.version: 2.3.2
rh-pre-commit.check-secrets: ENABLED

lcarva commented on installer/charts/rhtap-acs-scanner/scripts/test-acs-image-scan.sh:
```diff
@@ -0,0 +1,46 @@
+export ROX_CENTRAL_ENDPOINT
+export ROX_API_TOKEN
+
+IMAGE="${IMAGE:-quay.io/fedora/fedora:36-x86_64@sha256:d6e4c7d6d1eaa24d71c8efd7432890acdc0179502224d0aaad6bb05d15ffde19}"
```
Oh?

lcarva replied:
I understand it's only used for testing the installation, but can we use an image from registry.redhat.io instead?

Roming22 replied:
Unfortunately that's more complicated than just switching the image, as ACS would need to be setup to be integrated with registry.redhat.io (it's basically currently returning an error saying it can't get the image from that registry).

I've chosen as close to possible to a "Red Hat" image by using a Fedora image hosted on quay, which are 2 products strongly linked to our brand. Let me know if you don't think that's good enough.

lcarva replied:
Thanks for looking into it! I think it's fine because:

1. The contents of the fedora image are never executed.
2. The image comes from a registry that is managed by Red Hat.

otaviof commented on installer/charts/rhtap-acs-scanner/scripts/test-acs-image-scan.sh:
```diff
@@ -0,0 +1,44 @@
+set -o pipefail
+
+ROX_CENTRAL_ENDPOINT="$(kubectl get secrets -n rhtap rhtap-acs-integration -o jsonpath='{.data.endpoint}' | base64 -d)"
+ROX_API_TOKEN="$(kubectl get secrets -n rhtap rhtap-acs-integration -o jsonpath='{.data.token}' | base64 -d)"
```
You need to mount this secret on the testing POD, by doing so the `ROX_CENTRAL_ENDPOINT` and `ROX_API_TOKEN` will be available directly in the environment.

Roming22 replied:
Fixed

otaviof commented on installer/charts/rhtap-acs-scanner/scripts/test-acs-image-scan.sh:
```diff
@@ -0,0 +1,44 @@
+  echo "Retrying"
+done
+rm ./roxctl
+echo
```
Please use a `emptyDir: {}` mount, so at the end of the container execution all data in the mount point will be gone.

Roming22 replied:
Fixed

otaviof commented on installer/charts/rhtap-acs-scanner/templates/service-account.yaml:
```diff
@@ -0,0 +1,46 @@
+  namespace: {{ .Release.Namespace }}
+
+---
+apiVersion: rbac.authorization.k8s.io/v1
```
Does it need "write" permission to Secrets?

Roming22 replied:
In fact, we don't need the SA at all.
I now use the `default` SA in the `rhtap` namespace.

otaviof commented on installer/charts/rhtap-acs/templates/tests/test.yaml:
```diff
@@ -32,6 +32,11 @@ spec:
         - name: RESOURCE_TYPE
           value: "deployment"
+      {{- if eq . "scanner" }}
+        # Downloading the vulnerability DB can take a long time
```
By default we allow a 15 minute timeout. Should we make sure the Helm Chart installation waits longer than that?

Roming22 replied:
My tests shows that 15m is enough for the test to pass, we should be OK.

Roming22 replied:
Another test shows that the 15m Helm timeout does not apply to tests. I'm surprised by that result. There is a timeout flag (https://helm.sh/docs/helm/helm_test/#options), I wonder if we use it at all.

otaviof commented on installer/charts/rhtap-acs-scanner/scripts/test-acs-image-scan.sh:
This script should live on the `installer/scritps` directory, with a symbolic link into the Helm Chart directory, following the approach taken by all other scripts. Please move it.

Roming22 replied:
Fixed

Roming22 commented:
/retest

Roming22 committed:
fix: Enable scannerV4 in ACS

Recommended configuration after reaching out the ACS team regarding
RHTAPBUGS-1316.

rh-pre-commit.version: 2.3.2
rh-pre-commit.check-secrets: ENABLED

otaviof reviewed (approved):
/lgtm
